lib/devices/set_ntp.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Onvif.Devices.SetNTP do
+  import SweetXml
+  import XmlBuilder
+
+  alias Onvif.Device
+  alias Onvif.Devices.NTP
+
+  def soap_action, do: "http://www.onvif.org/ver10/device/wsdl/SetNTP"
+
+  @spec request(Device.t(), list) :: {:ok, any} | {:error, map()}
+  def request(device, args) do
+    Onvif.Devices.request(device, args, __MODULE__)
+  end
+
+  def request_body(config: %NTP{} = ntp) do
+    element(:"s:Body", [
+      element(:"tds:SetNTP", [
+        element(:"tds:FromDHCP", ntp.from_dhcp),
+        ntp_manual_element(ntp, ntp.from_dhcp) |> List.flatten()
+      ])
+    ])
+  end
+
+  defp ntp_manual_element(%NTP{} = _ntp, true), do: []
+
+  defp ntp_manual_element(%NTP{} = ntp, false) do
+    [element(:"tds:NTPManual", ntp_add_manual_element(ntp.ntp_manual))]
+  end
+
+  defp ntp_add_manual_element(ntp_manual) do
+    [
+      element(:"tt:Type", ntp_manual.type),
+      ntp_manual_element_data(ntp_manual)
+    ]
+  end
+
+  defp ntp_manual_element_data(ntp_manual) do
```

Comment:
```suggestion
  defp ntp_manual_element_data(%NTPManual{type: ipv4}) do
    element(:"tt:IPv4Address", ntp_manual.ipv4_address)
  end
```

nit: pattern matching instead of conditionals for all the different types